app/services/katello/registration_manager.rb

```diff
@@ -234,7 +234,7 @@ def set_host_collections(host, activation_keys)
         host_collection_ids.each do |host_collection_id|
           host_collection = ::Katello::HostCollection.find(host_collection_id)
           if !host_collection.unlimited_hosts && host_collection.max_hosts >= 0 &&
-             host_collection.systems.length >= host_collection.max_hosts
+             host_collection.hosts.length >= host_collection.max_hosts
```

Comment:
@jlsherrill @jturel The [fixture changes](https://github.com/Katello/katello/pull/9705/files#diff-106fbd1ffc78c4fbe1547d5a606f22a2a607dae9ee79060bce74c033cb2a32c7) in the previous PR uncovered this code path which apparently didn't have test coverage.

```
Error:

[2021-12-03T14:33:01.224Z] Katello::Service::RegistrationManager#test_registration_activation_key:
[2021-12-03T14:33:01.224Z] NoMethodError: undefined method `systems' for #<Katello::HostCollection:0x000000000fcd58c0>
[2021-12-03T14:33:01.224Z]     /home/jenkins/workspace/katello-pr-test@2/app/services/katello/registration_manager.rb:237:in `block in set_host_collections'
[2021-12-03T14:33:01.224Z]     /home/jenkins/workspace/katello-pr-test@2/app/services/katello/registration_manager.rb:234:in `each'
[2021-12-03T14:33:01.224Z]     /home/jenkins/workspace/katello-pr-test@2/app/services/katello/registration_manager.rb:234:in `set_host_collections'
[2021-12-03T14:33:01.224Z]     /home/jenkins/workspace/katello-pr-test@2/app/services/katello/registration_manager.rb:156:in `register_host'
[2021-12-03T14:33:01.224Z]     /home/jenkins/workspace/katello-pr-test@2/test/services/katello/registration_manager_test.rb:226:in `test_registration_activation_key'
[2021-12-03T14:33:01.224Z]     /home/jenkins/workspace/katello-pr-test@2/test/support/vcr.rb:26:in `block in run'
[2021-12-03T14:33:01.224Z]     /home/jenkins/workspace/katello-pr-test@2/test/support/vcr.rb:25:in `run'
```

I see nowhere else in our code base that implies that there should be a `.systems` method. It seems like the host collection limit was never being enforced. Am I missing something?

Reply:
I guess this means NO ONE is assigning host collections to registration keys because they are certainly registering hosts with activation keys! Very interesting.

I think we need a new test in there to cover this path with 'exceeds maximum usage limit' message. Recommend using a factory rather than a fixture in this case since we need a particular setup on the collection (or a new fixture?). Refer to `test_registration_activation_key` as a template for the new test